client/app/lib/onboarding.coffee

```diff
@@ -46,11 +54,15 @@ module.exports = class Onboarding
         throw new Error 'Missing mandatory `steps` parameter' unless steps
 
         @user = user
-        @steps = steps.map (step) =>
-            stepModel = new Step step
-            stepModel.onValidated () =>
-                @handleStepSubmitted()
-            return stepModel
+        @steps = steps
+            .reduce (activeSteps, step) =>
+                stepModel = new Step step
+                if stepModel.isActive user
+                    activeSteps.push stepModel
+                    stepModel.onValidated () =>
+                        @handleStepSubmitted()
```

Comment:
Good question, I asked it to myself. Actually if we do this like it, we lose the onboading `this` context, as CoffeeScript does not offer a way to directly bind a function (or maybe I just don't know how to do). Wrapping this code in another function is the only way I found keep this context.